src/Http/Provider/AbstractHttpProvider.php

```diff
@@ -106,9 +106,9 @@ protected function getParsedResponse(RequestInterface $request): string
     /**
      * Returns the HTTP adapter.
      *
-     * @return HttpClient
+     * @return ClientInterface
      */
-    protected function getHttpClient(): HttpClient
+    protected function getHttpClient(): ClientInterface
```

Comment:
We are technically widening the return type. If someone extended this, they will get an error, right? Just as the `DummyProvider`.

Reply:
On PHP 7.3 they will get an error yes, but it is allowed as of PHP 7.4.

Reply:
Support for PHP 7.3 has been dropped (since it has reached end of life) (see #1158)